### ISSUE TITLE
net.openid.appauth.RedirectUriReceiverActivity Crash

### ISSUE THREAD

**Issue by yinyerong** [openid/AppAuth-Android#575]
java.lang.RuntimeException: Unable to start activity ComponentInfo{XXX/net.openid.appauth.RedirectUriReceiverActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.huawei.intentfuzzer.fuzz.TestParcelable
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3476)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3648)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:86)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:117)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:74)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2209)
    at android.os.Handler.dispatchMessage(Handler.java:112)
    at android.os.Looper.loop(Looper.java:295)
    at android.app.ActivityThread.main(ActivityThread.java:7586)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.huawei.intentfuzzer.fuzz.TestParcelable
    at android.os.Parcel.readParcelableCreator(Parcel.java:2845)
    at android.os.Parcel.readParcelable(Parcel.java:2771)
    at android.os.Parcel.readValue(Parcel.java:2674)
    at android.os.Parcel.readArrayMapInternal(Parcel.java:3043)
    at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:288)
    at android.os.BaseBundle.unparcel(BaseBundle.java:232)
    at android.os.BaseBundle.containsKey(BaseBundle.java:508)
    at android.content.Intent.hasExtra(Intent.java:7416)
    at android.app.Activity.finish(Activity.java:5966)
    at android.app.Activity.finish(Activity.java:5978)
    at net.openid.appauth.RedirectUriReceiverActivity.onCreate(:56)
    at android.app.Activity.performCreate(Activity.java:7507)
    at android.app.Activity.performCreate(Activity.java:7497)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1280)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
    ... 11 more

**Comment by santipg382s** [comment]
> java.lang.RuntimeException: Unable to start activity ComponentInfo{XXX/net.openid.appauth.RedirectUriReceiverActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.huawei.intentfuzzer.fuzz.TestParcelable
> at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3476)
> at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3648)
> at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:86)
> at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:117)
> at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:74)
> at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2209)
> at android.os.Handler.dispatchMessage(Handler.java:112)
> at android.os.Looper.loop(Looper.java:295)
> at android.app.ActivityThread.main(ActivityThread.java:7586)
> at java.lang.reflect.Method.invoke(Native Method)
> at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
> at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
> Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.huawei.intentfuzzer.fuzz.TestParcelable
> at android.os.Parcel.readParcelableCreator(Parcel.java:2845)
> at android.os.Parcel.readParcelable(Parcel.java:2771)
> at android.os.Parcel.readValue(Parcel.java:2674)
> at android.os.Parcel.readArrayMapInternal(Parcel.java:3043)
> at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:288)
> at android.os.BaseBundle.unparcel(BaseBundle.java:232)
> at android.os.BaseBundle.containsKey(BaseBundle.java:508)
> at android.content.Intent.hasExtra(Intent.java:7416)
> at android.app.Activity.finish(Activity.java:5966)
> at android.app.Activity.finish(Activity.java:5978)
> at net.openid.appauth.RedirectUriReceiverActivity.onCreate(:56)
> at android.app.Activity.performCreate(Activity.java:7507)
> at android.app.Activity.performCreate(Activity.java:7497)
> at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1280)
> at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3455)
> ... 11 more

**Comment by blundell** [comment]
Please create a Minimal Reproducible Example of the problem you are having, so others can validate/investigate.

**Comment by yinyerong** [comment]
> Please create a Minimal Reproducible Example of the problem you are having, so others can validate/investigate.
This is the Monkey test.

**Comment by mayumipal** [comment]
Fatal Exception: java.lang.RuntimeException
Unable to start activity ComponentInfo{com.microimage.hcmv3/net.openid.appauth.RedirectUriReceiverActivity}: java.lang.RuntimeException: Parcel android.os.Parcel@d5e9d29: Unmarshalling unknown type code 6881389 at offset 92
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2583)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2665)
android.app.ActivityThread.-wrap11 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1499)
android.os.Handler.dispatchMessage (Handler.java:111)
android.os.Looper.loop (Looper.java:207)
android.app.ActivityThread.main (ActivityThread.java:5765)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)
Caused by java.lang.RuntimeException
Parcel android.os.Parcel@d5e9d29: Unmarshalling unknown type code 6881389 at offset 92
android.os.Parcel.readValue (Parcel.java:2319)
android.os.Parcel.readArrayMapInternal (Parcel.java:2592)
android.os.BaseBundle.unparcel (BaseBundle.java:221)
android.os.Bundle.getBundle (Bundle.java:763)
com.google.firebase.messaging.FcmLifecycleCallbacks.onActivityCreated (com.google.firebase:firebase-messaging@@23.0.0:4)
android.app.Application.dispatchActivityCreated (Application.java:195)
android.app.Activity.onCreate (Activity.java:928)
androidx.core.app.ComponentActivity.onCreate (ComponentActivity.java:85)
androidx.activity.ComponentActivity.onCreate (ComponentActivity.java:323)
androidx.fragment.app.FragmentActivity.onCreate (FragmentActivity.java:273)
net.openid.appauth.RedirectUriReceiverActivity.onCreate (RedirectUriReceiverActivity.java:49)
android.app.Activity.performCreate (Activity.java:6319)
android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1113)
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2530)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2665)
android.app.ActivityThread.-wrap11 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1499)
android.os.Handler.dispatchMessage (Handler.java:111)
android.os.Looper.loop (Looper.java:207)
android.app.ActivityThread.main (ActivityThread.java:5765)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)

**Comment by adindesaputhran** [comment]
ActivityTaskManager : START u0 {act=android.intent.action.VIEW cat=[android.intent.category.BROWSABLE] flg=0x14000000 hwFlg=0x10 cmp=com.bpost.imove.dev/net.openid.appauth.RedirectUriReceiverActivity (has extras)} from uid 10089
2022-05-26 20:51:58.344	INFO:	ActivityTaskManager : ActivityRecord info: ActivityInfo{76ebf22 net.openid.appauth.RedirectUriReceiverActivity}
2022-05-26 20:51:58.344	INFO:	HwActivityTaskManagerServiceEx : packageName:com.bpost.imove.dev,requestCode:-1,isIntercepted:false,userId:0
2022-05-26 20:51:58.344	INFO:	HwActivityTaskManagerServiceEx : callingPackage:com.android.chrome
2022-05-26 20:51:58.344	INFO:	HwActivityTaskManagerServiceEx : is not cold start
2022-05-26 20:51:58.344	INFO:	HwActivityTaskManagerServiceEx : is not cold start
2022-05-26 20:51:58.345	DEBUG:	ActivityTaskManager : TaskLaunchParamsModifier:task=null activity=ActivityRecord{eafedb3 u0 com.bpost.imove.dev/net.openid.appauth.RedirectUriReceiverActivity t-1} display-from-source=0 display-id=0 display-windowing-mode=1
2022-05-26 20:51:58.345	DEBUG:	ActivityTaskManager : TaskLaunchParamsModifier:task=TaskRecord{b66052a #1062 A=com.bpost.imove.dev U=0 StackId=11 sz=3} activity=ActivityRecord{eafedb3 u0 com.bpost.imove.dev/net.openid.appauth.RedirectUriReceiverActivity t-1} display-from-task=0 display-id=0 display-windowing-mode=1 activity-options-fullscreen=Rect(0, 0 - 0, 0) inherit-fullscreen non-freeform-display
2022-05-26 20:51:58.345	VERBOSE:	ActivityTaskManager : positionChild stackId=11 to top.
2022-05-26 20:51:58.346	INFO:	app_process : Background concurrent copying GC freed 5283452(188MB) AllocSpace objects, 2(112KB) LOS objects, 17% free, 115MB/139MB, paused 58us total 616.969ms
2022-05-26 20:51:58.348	DEBUG:	AwareLog : iawared:  decompress gpu mem, pid:802
2022-05-26 20:51:58.346	VERBOSE:	ActivityTaskManager : positionChild stackId=11 to top.
2022-05-26 20:51:58.350	VERBOSE:	ActivityTaskManager : addAppToken: AppWindowToken{4842e9 token=Token{6c6ef70 ActivityRecord{eafedb3 u0 com.bpost.imove.dev/net.openid.appauth.RedirectUriReceiverActivity t1062}}} task={taskId=1062 appTokens=[AppWindowToken{4e391c3 token=Token{f92972 ActivityRecord{732a67d u0 com.bpost.imove.dev/com.bpost.imove.onboarding.OnboardingContainerActivity t1062}}}, AppWindowToken{fbfa7a6 token=Token{cf02d01 ActivityRecord{44252e8 u0 com.bpost.imove.dev/net.openid.appauth.AuthorizationManagementActivity t1062}}}, AppWindowToken{2275044 token=Token{5ccbd57 ActivityRecord{dec36d6 u0 com.android.chrome/org.chromium.chrome.browser.customtabs.CustomTabActivity t1062}}}] mdr=false} at 2147483647
2022-05-26 20:51:58.351	INFO:	PG_ash : front pkg : com.bpost.imove.dev launcher: false
2022-05-26 20:51:58.351	INFO:	PG_ash : above launcher front pkgs: [com.bpost.imove.dev]
2022-05-26 20:51:58.371	WARN:	AccessibilityInteractionClient : old interaction Id is: -1,current interaction Id is:5528
2022-05-26 20:51:58.380	WARN:	ImageReader_JNI : Unable to acquire a buffer item, very likely client tried to acquire more than maxImages buffers
2022-05-26 20:51:58.383	INFO:	PGServer : android calls fast HApps
2022-05-26 20:51:58.383	INFO:	AppFastHibernation : fast H duration: 1500 reason: start activity
2022-05-26 20:51:58.384	INFO:	Pged-IPgedBinderService : ****action is 1, pid num is 3***
2022-05-26 20:51:58.384	INFO:	Pged-Freezer : freezePids
2022-05-26 20:51:58.384	INFO:	Pged-Freezer : Freeze process: 32422
2022-05-26 20:51:58.385	INFO:	Pged-Freezer : Freeze process: 31263
2022-05-26 20:51:58.386	DEBUG:	PgedBinderAdapter : F_Z ok, pid: [32422, 31263]
2022-05-26 20:51:58.389	INFO:	ActivityManager : Killing 3254:com.android.chrome:sandboxed_process0:org.chromium.content.app.SandboxedProcessService0:1/u0a89i-8999 (adj 0): isolated not needed
2022-05-26 20:51:58.391	INFO:	libprocessgroup : Successfully killed process cgroup uid 90001 pid 3254 in 0ms
2022-05-26 20:51:58.399	DEBUG:	HwConnectivityServiceEx : set cache by key is 10089 interfaceName iswlan0 value is false
2022-05-26 20:51:58.399	DEBUG:	ConnectivityService : get Network state from cache, cacheBlock is 0
2022-05-26 20:51:58.400	WARN:	ActivityManager : Sending non-protected broadcast huawei.action.floattasks.IME_WINDOW_STATE_CHANGED from system 1768:com.android.systemui/u0a32 pkg com.android.systemui

**Comment by adindesaputhran** [comment]
i have the same issue in huawei p30 lite

**Comment by adindesaputhran** [comment]
is there any workaround for this crash?

**Comment by SPodjasek** [comment]
Nevertheless, that issue is already closed - I've decided to post my solution here as this issue was one of the first I've found investigating this problem. My Flutter Android app was working perfectly with AppAuth to the point where I added local_auth - then this error appeared.
The solution was to include `androidx.appcompat:appcompat` in dependencies (`./android/app/build.gradle`) and changing styles from `@android:style/Theme.*` to `Theme.AppCompat.*`.
I hope that somebody will find this hint helpful.
